### PR TITLE
Issue-12: Add initial Docusaurus documentation

### PR DESCRIPTION
**Pull request by DanielJette**
### What does this change accomplish?

This PR adds the initial version of the Docusaurus documentation. The documentation is a direct port of the documentation currently found in markdown files.

These documents represent a snapshot of the 1.2.0-alpha01 documentation and will eventually be versioned as such and replaced with updated 2.* documentation.

Please review https://ndtp.github.io/android-testify/ and report broken links or any navigation problems.

### CODE REVIEW

**Review by lwise — CHANGES_REQUESTED** [review]
We should update the references to the old Shopify plugin. Also took the opportunity to look at the docs with fresh eyes.

**Review by lwise — COMMENTED** [review]
\*Reads the PR description\*
> These documents represent a snapshot of the 1.2.0-alpha01 documentation and will eventually be versioned as such and replaced with updated 2.* documentation.

It might be confusing to have out-of-sync docs up - we should at least clarify within the docs that these are for an older version of Testify until they're updated.

File: docs/docs/intro.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+---
+sidebar_position: 1
+---
+
+# Android Screenshot Testing
+
+## Testify
+
+Expand your test coverage by including the View-layer. Testify allows you to easily set up a variety of screenshot tests in your application. Capturing a screenshot of your view gives you a new tool for monitoring the quality of your UI experience. It's also an easy way to review changes to your UI. Once you've established a comprehensive set of screenshots for your application, you can use them as a "visual dictionary". In this case, a picture really is worth a thousand words; it's easy to catch unintended changes in your view rendering by watching for differences in your captured images.
```
[the remark this answers is not in the document]
Comment (DanielJette) [reply docs/docs/intro.md:9]: Yeah, that looks silly. Will change.

File: docs/docs/get-started/1-setup.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+Before building your screenshot test with Testify, make sure to set a dependency reference to the Testify plugin:
+
+**Root build.gradle**
+```groovy
+buildscript {
+    repositories {
+        mavenCentral()
+    }
+    dependencies {
+        classpath "com.shopify.testify:plugin:1.2.0-alpha01"
```
[the remark this answers is not in the document]
Comment (DanielJette) [reply docs/docs/get-started/1-setup.md:12]: @lwise This is a direct port of the Shopify documentation and I'm going to version these documents as 1.2.0-alpha01 in https://github.com/ndtp/android-testify/pull/76

So, technically this is correct for what I wanted right now. I'll update this to `dev.testify` stuff when I publish the 2.0.0-alpha01 documentation.

File: docs/docs/get-started/2-configuring-an-emulator.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+- RAM: 1536 MB
+- VM heap: 256 MB
+- Internal Storage: 2048 MB
+- SD card, Studio-managed: 512 MB
+- Enable Device Frame with pixel_3a skin
+- Enable keyboard input
+
+Once the emulator is booted:
+- Set the Language to English (United States) (`en_US`)
+- In the developer settings, set Window animation scale, Transition animation scale, and Animator duration scale to `off`
```
[the remark this answers is not in the document]
Comment (DanielJette) [reply docs/docs/get-started/2-configuring-an-emulator.md:16]: Good idea. I'll add them to the 2.0 documentation

File: docs/docs/recipes/11-layout-inspector.md [hunk 4]
```diff
@@ -0,0 +1,13 @@
+# Debugging with the Layout Inspector
+
+You may use Android Studio's [Layout Inspector](https://developer.android.com/studio/debug/layout-inspector) in conjunction with your screenshot test. It can sometimes be useful to pause your test so that you can capture the layout hierarchy for further debugging in Android Studio. In order to do so, invoke the `setLayoutInspectionModeEnabled` method on the test rule. This will pause the test after all ViewModifications have been applied and prior to the screenshot being taken. The test is paused for 5 minutes, allowing plenty of time to capture the layout.
+
+```kotlin
+    @ScreenshotInstrumentation
+    @Test
+    fun testDefault() {
+        rule
+                .setLayoutInspectionModeEnabled(true)
```
[the remark this answers is not in the document]
Comment (DanielJette) [reply docs/docs/recipes/11-layout-inspector.md:10]: 🤔 It might be already. I'll check the code.
Otherwise, I can add that to 2.0

**Comment by DanielJette** [comment]
> *Reads the PR description*
> 
> > These documents represent a snapshot of the 1.2.0-alpha01 documentation and will eventually be versioned as such and replaced with updated 2.* documentation.
> 
> It might be confusing to have out-of-sync docs up - we should at least clarify within the docs that these are for an older version of Testify until they're updated.

@lwise Yeah, the only reason I did it this way is that I want to have all the versions documented using the versioning system for Docusaurus. For that, I've started by porting the Shopify docs over, then freezing those to the 1.2.0-alpha01 version. I have PRs queued up in draft for the follow-up documentation to make the current docs reflect the latest version. So, it's just a matter of getting everything merged so it makes sense.

**Review by lwise — APPROVED** [review]
Nice, I didn't realize docusaurus supported a versioning system like that

**Pull request merged**